routing/pathfind.go

```diff
@@ -415,44 +441,22 @@ func findPath(g *graphParams, r *RestrictParams, cfg *PathFindingConfig,
 	}()
 
 	// Get source node outside of the pathfinding tx, to prevent a deadlock.
-	selfNode, err := g.graph.SourceNode()
-	if err != nil {
-		return nil, err
-	}
-	self := selfNode.PubKeyBytes
-
-	// Get a db transaction to execute the graph queries in.
-	tx, err := g.graph.Database().Begin(false)
-	if err != nil {
-		return nil, err
-	}
-	defer tx.Rollback()
+	self := graph.sourceNode()
```

Comment:
above comment is no longer accurate and should be moved to where we actually fetch the source node

Reply:
comment removed and call moved down to where it is first used